File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    void backtrack(int index, vector<int>& nums, vector<int>& current, vector<vector<int>>& result) {
        // Add the current subset to the result
        result.push_back(current);

        // Explore further elements
        for (int i = index; i < nums.size(); i++) {
            current.push_back(nums[i]);               // include nums[i]
            backtrack(i + 1, nums, current, result);  // move to next index
            current.pop_back();                       // backtrack
        }
    }

    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> result;
        vector<int> current;
        backtrack(0, nums, current, result);
        return result;
    }
};

File: 0078-subsets/0078-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    let result=[];
    let current=[];
    
    function backtrack(start){
        result.push([...current]);

        for(let i=start; i<nums.length; i++){
            current.push(nums[i])
            backtrack(i+1)
            current.pop()
        }
    }


    backtrack(0);
    return result;
};